.github/workflows/job-precommit-java-pvr-flink-batch.yml

```diff
@@ -0,0 +1,85 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# To learn more about GitHub Actions in Apache Beam check the CI.md
+
+# Runs PreCommit Java PVR Flink Batch scheduled on cron, on push to master and on pull request
+
+name: PreCommit Java PVR Flink Batch
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 */6 * * *'
+  push:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+  pull_request_target:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+    paths: ['sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/**',
+            'runners/flink/**',
+            'runners/java-fn-execution/**']
+permissions: read-all
+
+
+jobs:
+  set-properties:
+    runs-on: [self-hosted, ubuntu-20.04]
+    outputs:
+      properties: ${{ steps.test-properties.outputs.properties }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          persist-credentials: false
+          submodules: false
+      - id: test-properties
+        uses: ./.github/actions/setup-default-test-properties
+
+  precommit-java-pvr-flink-batch:
+    needs: set-properties
+    name: Run PreCommit Java PVR Flink Batch
+    runs-on: [self-hosted, ubuntu-20.04]
+    strategy:
+      fail-fast: false
+      matrix: #JDK17 execution will fail due to new access policies applied to the core JDK libraries starting at JDK 16.
+        java: ${{fromJson(needs.set-properties.outputs.properties).JavaTestProperties.SUPPORTED_VERSIONS}}
+        flink: ${{fromJson(needs.set-properties.outputs.properties).JavaTestProperties.FLINK_VERSIONS}}
+        exclude: #TODO: Remove exclude and retest once issue https://github.com/apache/beam/issues/23066 gets resolved.
+          - java: "11" #Creation of the DockerEnvironment fails with Java 11.
```

Comment:
Is there an issue tracking this?

Reply:
No issue for this one yet, I will go ahead and create one.

Reply:
Deleted the previous comment. Just realized the problem with the creation of the DockerEnvironment is due to the ErrorProne library issues when compiled with JDK11. The issue tracking that is this one: https://github.com/apache/beam/issues/23066